src/main/java/org/openrewrite/polyglot/RemoteProgressBarSender.java

```diff
@@ -93,6 +91,13 @@ private void send(Request.Type type, @Nullable String message) {
         }
     }
 
+    private static @Nullable String truncateMessage(@Nullable String message, int maxLength) {
+        if (message == null || message.length() <= maxLength) {
+            return message;
+        }
+        return "..." + message.substring(Math.max(message.length() - maxLength - 3, 0));
```

Comment:
This could still fail (on the receiving side) if there are any non-ASCII characters requiring a multibyte encoding. But that is rather unlikely case, so I guess we can ignore that for now. Otherwise, we would have to truncate the encoding instead. To do that properly, we would have to check that we don't accidentally truncate the byte array in the middle of a multibyte encoding (and if we do, then truncate all those bytes part of that multibyte set as well). For UTF-8 I believe we can do that by checking if the value is greater than `0xBF`.